pkg/storaged/block/format-dialog.jsx

```diff
@@ -631,18 +621,24 @@ function format_dialog_internal(client, path, start, size, enable_dos_extended,
                     return client.blocks_crypto[path];
                 }
 
-                function maybe_mount(new_path) {
+                async function maybe_mount(new_path) {
                     const path = new_path || block.path;
-                    if (is_filesystem(vals) && mount_now)
-                        return (client.wait_for(() => block_fsys_for_block(path))
-                                .then(block_fsys => client.mount_at(client.blocks[block_fsys.path],
-                                                                    mount_point)));
-                    if (type == "swap" && mount_now)
-                        return (client.wait_for(() => block_swap_for_block(path))
-                                .then(block_swap => block_swap.Start({})));
-                    if (is_encrypted(vals) && (is_filesystem(vals) || type == "swap") && !mount_now)
-                        return (client.wait_for(() => block_crypto_for_block(path))
-                                .then(block_crypto => block_crypto.Lock({ })));
+                    const new_block = await client.wait_for(() => client.blocks[path]);
+
+                    if (is_encrypted(vals))
+                        remember_passphrase(new_block, vals.passphrase);
+
+                    if (is_filesystem(vals) && mount_now) {
+                        await client.mount_at(new_block, mount_point);
+                    }
+                    if (type == "swap" && mount_now) {
+                        const block_swap = await client.wait_for(() => block_swap_for_block(path));
+                        await block_swap.Start({});
+                    }
+                    if (is_encrypted(vals) && (is_filesystem(vals) || type == "swap") && !mount_now) {
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19924-20240207-125230-6cf61ca9-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)